File: election-ui-swift/Modules/ElectionResults/Sources/ElectionResults/Public/ResultsServiceFactory.swift

import Foundation

/// Factory to create a new `ResultsService` configured for use.
public class ResultsServiceFactory {
    
    /// Create a new `ResultsService` object
    /// - Returns: the configured `ResultsService` ready for use
    public static func resultsService() -> ResultsService {
        ResultsServiceImpl(resultsRepository: JSONResultsRepository())
    }
}

File: election-ui-kotlin/api/src/main/java/uk/co/bbc/elections/api/ResultsServiceFactory.kt
package uk.co.bbc.elections.api

import uk.co.bbc.elections.api.internal.JSONResultsRepository

/**
 * Factory to create a new ResultsService ready for use.
 */
class ResultsServiceFactory {
    companion object {
        /**
         * Build a new ResultsService.
         *
         * @return the configured ResultsService ready for use.
         */
        fun buildResultsService(): ResultsService {
            return ResultsServiceImpl(JSONResultsRepository())
        }
    }
}